util/src/test/scala/org/broadinstitute/dsde/rawls/util/RetrySpec.scala

```diff
@@ -0,0 +1,179 @@
+package org.broadinstitute.dsde.rawls.util
+
+import akka.actor.ActorSystem
+import akka.testkit.TestKit
+import com.typesafe.scalalogging.{LazyLogging, Logger}
+import org.mockito.ArgumentCaptor
+import org.mockito.ArgumentMatchers._
+import org.mockito.Mockito._
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.mock.MockitoSugar
+import org.scalatest.time.{Minutes, Seconds, Span}
+import org.scalatest.{BeforeAndAfterAll, FlatSpecLike, Matchers}
+import org.slf4j.{Logger => Underlying}
```

Comment:
To deal with name conflicts between `org.slf4j.Logger` and `com.typesafe.scalalogging.Logger`. `Retry` mixes in `LazyLogging` which uses `com.typesafe.scalalogging.Logger`. But it turns out `com.typesafe.scalalogging.Logger` is not mockable because it's a `final` class, so I'm mocking the underlying `org.slf4j.Logger` instead and passing it to the `com.typesafe.scalalogging.Logger` constructor:
https://github.com/typesafehub/scalalogging/blob/master/scalalogging-slf4j/src/main/scala/com/typesafe/scalalogging/slf4j/Logger.scala#L38

I agree it's confusing, maybe I'll rename the type to `SLF4JLogger` for clarity.